java/core/src/main/java/com/microsoft/bond/BlobBondType.java

```diff
@@ -72,25 +72,25 @@ public final boolean isGenericType() {
     }
 
     @Override
-    protected final byte[] newDefaultValue() {
+    protected final Blob newDefaultValue() {
         return DEFAULT_VALUE;
     }
 
     @Override
-    protected final byte[] cloneValue(byte[] value) {
+    protected final Blob cloneValue(Blob value) {
         return value.clone();
     }
```

Comment:
This should return a deep clone, which is not what Blob.clone does. Please see the API of cloneValue in BondType.